fiftyone/core/dataset.py

```diff
@@ -6391,7 +6391,7 @@ def _load_clips_source_dataset(frame_collection_name):
     return load_dataset(doc["name"])
 
 
-def _load_dataset(obj, name, virtual=False):
+def _load_dataset(obj, name, virtual=False, slug=False):
```

Comment:
I **don't** think we should allow datasets to be loaded by slug in the Python SDK.

There are quite a few places where dataset names are used as identifiers of datasets (eg listing, deleting) not captured here, so this change is incomplete, but regardless I don't see any benefit to allowing slug-based access; it will likely just lead to confusion.

Slug is not currently a user-facing concept, its just an internal way to generate human readable URLs.

Reply:
I wasn't planning on exposing it and forgot to remove this arg. 

Just to clarify, do you mean you don't want to include `slug` in the `find` query? Not sure I understand the reasoning why using slugs in the query would be problematic since slugs have to be unique so the lookup should only result in 1 dataset. If not, there is a problem elsewhere.

Alternatively, would you prefer me to write a separate function that first queries the db for a dataset name given a slug? And then load the dataset with that name? The issue is that given a direct link/slug to a dataset, there is no way of deriving the name without an additional lookup since the relationship between unique slugs and unique names is 1: many.